### PR TITLE
Adding turn icons API to expose getting the drawables from TurnIconHe…

### LINKED ISSUE #4811: Expose turn icon drawables
Similar to how we exposed the lane icons https://github.com/mapbox/mapbox-navigation-android/pull/4729

We should expose the turn icons. Android Auto needs to duplicate the sdk logic in order to extract the maneuver icons
https://github.com/mapbox/mapbox-navigation-android-examples/blob/60d3132bda3f8fb07bce51fc0dccff2fc2d6f6ce/android-auto/src/main/java/com/mapbox/examples/androidauto/car/navigation/CarManeuverMapper.kt#L49-L79

~If we're too late for SEMVER changes in rc-1.0.0, then we can copy the sdk into the android auto solution. Roundabout icons were added after the first android auto prototype https://github.com/mapbox/mapbox-navigation-android/pull/4683 (i think 🤔 )~

cc: @cafesilencio @abhishek1508 @Guardiola31337 @pengdev

### PR DESCRIPTION
**Pull request by cafesilencio**
### Description
Resolves #4811 by adding API class to expose the functionality in `TurnIconHelper` for android auto support.

### Changelog
```
<changelog>Added MapboxTurnIconsApi for retrieving turn icon drawables.</changelog>
```

### CODE REVIEW

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxTurnIconsApi.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * Returns a reference to a drawable for turn icons.
+     *
+     * @param type a maneuver type
+     * @param degrees a value describing the degree of turn
+     * @param modifier a [ManeuverModifier] value
+     * @param drivingSide the driving side describing the turn
+     */
+    fun generateTurnIcon(
```
Comment (kmadsen) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxTurnIconsApi.kt:28]: Will you also migrate the `MapboxTurnIconManeuver` class to use this API?

That way our primary use case can ensure this api works

Comment (cafesilencio) [reply libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxTurnIconsApi.kt:28]: sure

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxTurnIconsApi.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+     * @param resources TurnIconResources
+     */
+    fun updateResources(resources: TurnIconResources) {
+        turnIconHelper = TurnIconHelper(resources)
+    }
+
+    private fun generateError(type: String?, drivingSide: String?): TurnIconError {
+        val errorMessage = "Unrecognized turn $type drivingSide: $drivingSide"
+        return TurnIconError(errorMessage, type, drivingSide)
+    }
```
Comment (kmadsen) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxTurnIconsApi.kt:58]:
```suggestion
    private fun generateError(type: String?, degrees: Float?, modifier: String?, drivingSide: String?): TurnIconError {
        val errorMessage = "Unrecognized turn $type degrees: $degrees modifier: $modifier drivingSide: $drivingSide"
        return TurnIconError(errorMessage, type, degrees, modifier, drivingSide)
    }
```

Add all the arguments to the error. We have errors downstream that were caused by unknown "degrees" for example.

Comment (cafesilencio) [reply libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxTurnIconsApi.kt]: done

**Review by kmadsen — APPROVED** [review]

**Comment by cafesilencio** [comment]
This got cherry picked into the release branch.

**Comment by Guardiola31337** [comment]
> This got cherry picked into the release branch.

Removing the `needs-backporting` label then 👍

**Pull request merged**